Added GetProcessId()

Yothri opened the pull request:
I needed that function. You can get the process id of a process by specifying a process handle.

Commit:
Added GetProcessId()

AArnott commented on src/Kernel32/Kernel32.cs:
```diff
@@ -157,6 +157,14 @@ public static partial class Kernel32
+        /// <summary>
+        /// Retrieves the process identifier of the specified process.
+        /// </summary>
+        /// <param name="Process">A handle to the process. The handle must have the PROCESS_QUERY_INFORMATION or PROCESS_QUERY_LIMITED_INFORMATION access right. For more information, see <see cref="https://msdn.microsoft.com/de-de/library/windows/desktop/ms684880(v=vs.85).aspx"/></param>
```
You evidently can't use `<see cref` for a URL as it generates a compiler warning. I recommend you just drop the tag and leave the URL.

Yothri replied:
I removed that in my local project and just put the name of the site there instead of the full URL.
Can I somehow edit this pull request instead of creating new one? I am not familiar with Git/PullRequests yet.

AArnott reviewed (approved):
Thanks!

AArnott commented on src/Kernel32/Kernel32.cs:
```diff
@@ -157,6 +157,14 @@ public static partial class Kernel32
+        /// <param name="Process">A handle to the process. The handle must have the PROCESS_QUERY_INFORMATION or PROCESS_QUERY_LIMITED_INFORMATION access right. For more information, see <see cref="https://msdn.microsoft.com/de-de/library/windows/desktop/ms684880(v=vs.85).aspx"/></param>
+        /// <returns>The process identifier of the specified process.</returns>
+        [DllImport(api_ms_win_core_processthreads_l1_1_1)]
+        public static extern int GetProcessId(IntPtr Process);
```
You will need to add this method to the PublicAPI.Nonshipping.txt file for the project. There should be a code fix offered if you click the light bulb on this line to automatically make this change.

Yothri replied:
Light Bulb doesnt show anything useful up to me, NonShipping.txt does also not exist, do you mean the Unshipped.txt?

![Image of VS](https://i.imgur.com/2dif5Ct.png)

AArnott replied:
Close. But I think you need to change `IntPtr` to `System.IntPtr` in the unshipped.txt file.

Yothri replied:
Ahh, okay, hehe, thanks for your patience :D

AArnott replied:
No problem. I appreciate your contribution.

There's still the trailing `"` symbol that doesn't belong. 😲

Commit:
Added to Unshipped.txt the function.
Removed cref from the function as it generates compiler warning when using a URL in it.

Commit:
Added namespace for IntPtr.

Yothri commented:
@AArnott That seems to have worked now, but I got another question. I want to add DwmExtendFrameIntoClientArea in the DwmApi.

SigImp doesnt find this function, but since there is a HResult already defined that should be no problem on the return value. But what about the MARGINS structure which is defined in the PInvoke.UxTheme library. DwmApi doesnt reference this library so it doesnt know this structure. Could you give me a hint on what to do here?

AArnott commented on src/Kernel32/Kernel32.cs:
```diff
@@ -157,6 +157,14 @@ public static partial class Kernel32
+        /// <summary>
+        /// Retrieves the process identifier of the specified process.
+        /// </summary>
+        /// <param name="Process">A handle to the process. The handle must have the PROCESS_QUERY_INFORMATION or PROCESS_QUERY_LIMITED_INFORMATION access right. For more information, see Process Security and Access Rights."/></param>
```
You still have a trailing `"/>` within your param doc.

AArnott commented:
> what about the MARGINS structure which is defined in the PInvoke.UxTheme library. DwmApi doesnt reference this library so it doesnt know this structure.

Generally we approach such questions starting with this: "How is this handled in native code?" If DwmApi.dll clients typically import uxtheme headers, then it is reasonable for dwmapi.dll to reference uxtheme.dll.

Commit:
Removed the trailing "/> within the param doc.

Yothri commented:
Wtf am I dumb

Commit:
Removed the trailing "

AArnott commented:
thanks
